java/code/src/com/suse/manager/webui/controllers/maintenance/MaintenanceController.java

```diff
@@ -114,4 +150,30 @@ static void handleRescheduleResult(List<RescheduleResult> results, RescheduleStr
             }
         });
     }
+
+    /**
+     * Apply the timezone shift of the user selected timezone and return json string in Fullcalendar required format
+     *
+     * @param user the current user
+     * @param response the response
+     * @param events list of MaintenanceWindowData
+     * @return the result json string
+     */
+    private static String postprocessMaintenanceWindows(
+            User user , Response response, List<MaintenanceWindowData> events) {
+        ZoneId zoneId = ZoneId.of(user.getTimeZone().getOlsonName());
+        List<Map<String, String>> data = new ArrayList<>();
+        events.forEach(event ->
+                data.add(Map.of(
+                        "title", event.getName(),
+                        "start", ZonedDateTime.ofInstant(
+                                Instant.ofEpochMilli(event.getFromMilliseconds()), zoneId
+                        ).toString().split("\\[")[0],
+                        "end", ZonedDateTime.ofInstant(
+                                Instant.ofEpochMilli(event.getToMilliseconds()), zoneId
+                        ).toString().split("\\[")[0]
+                ))
+        );
+        return json(response, ResultJson.success(data));
```

Comment:
Another nitpick: maybe you can create a `...Json` DTO class with `title`, `start` and `end` properties to make the API signature more clear (even a nested class would be ok)

Reply:
Done